schemas/ispyb/stored_programs/sp_retrieve_fluorescence_spectrum_ids_for_sample.sql

```diff
@@ -0,0 +1,19 @@
+DELIMITER ;;
+CREATE OR REPLACE DEFINER=`ispyb_root`@`%` PROCEDURE `retrieve_fluorescence_spectrum_ids_for_sample`(p_id int unsigned)
+READS SQL DATA
+COMMENT 'Returns a multi-row result-set with the fluorescence spectrum ids for the given sample id.'
+BEGIN
+	IF p_id IS NOT NULL THEN
+
+	SELECT DISTINCT
+	  xfe.xfeFluorescenceSpectrumId
+	FROM BLSample bls
+	  INNER JOIN XFEFluorescenceSpectrum xfe ON bls.blSampleId = xfe.blSampleId
+	WHERE bls.blSampleId = p_id;
```

Comment:
I think the query can be rewritten (simplified) as: 

	SELECT
	  xfeFluorescenceSpectrumId
	FROM XFEFluorescenceSpectrum
	WHERE blSampleId = p_id;